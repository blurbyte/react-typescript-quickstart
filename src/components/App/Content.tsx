import styled from '@styles/styledComponents';

const Content = styled.section`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 20px;
`;

export default Content;
