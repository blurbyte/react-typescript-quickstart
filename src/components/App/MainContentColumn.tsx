import styled from '@styles/styledComponents';

const MainContentColumn = styled.section`
  grid-column: 1 / -2;
`;

export default MainContentColumn;
