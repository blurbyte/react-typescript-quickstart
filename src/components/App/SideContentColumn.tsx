import styled from '@styles/styledComponents';

const SideContentColumn = styled.section`
  grid-column: -2 / -1;
`;

export default SideContentColumn;
