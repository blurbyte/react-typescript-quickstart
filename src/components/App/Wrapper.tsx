import styled from '@styles/styledComponents';
import media from '@styles/media';

const Wrapper = styled.h1`
  background-color: blue;

  ${media.tablet`
    background-color: red;
  `};
`;

export default Wrapper;
