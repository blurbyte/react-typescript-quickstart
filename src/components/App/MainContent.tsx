import styled from '@styles/styledComponents';
import media from '@styles/media';

import Content from '../Content';

const Article = Content.withComponent('article');

const MainContent = styled(Article)`
  display: grid;
  grid-template-columns: repeat(2, 1fr) minmax(30rem, 1fr);
  grid-gap: 0 3rem;
  padding-bottom: 6rem;
  padding-top: 9rem;

  ${media.tablet`
    grid-template-columns: repeat(3, 1fr);
    padding-top: 6rem;
  `};
`;

export default MainContent;
