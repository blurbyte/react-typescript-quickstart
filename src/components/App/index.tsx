import * as React from 'react';

import Header from '../Header';
import MainContent from '../MainContent';
import Overview from '../Overview';
import Wrapper from './Wrapper';

// In this case Props and State are automatically initialized to {}
class App extends React.Component {
  render() {
    return (
      <Wrapper>
        <Header />
        <MainContent>
          <Overview />
        </MainContent>
      </Wrapper>
    );
  }
}

export default App;
