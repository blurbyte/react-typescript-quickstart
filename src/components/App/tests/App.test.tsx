import * as React from 'react';
import { shallow, mount } from 'enzyme';

import App from '../index';

describe('App', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
