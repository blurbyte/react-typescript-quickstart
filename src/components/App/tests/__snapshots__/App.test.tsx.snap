// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<styled.div>
  <Header />
  <Styled(styled.article)>
    <Overview />
  </Styled(styled.article)>
</styled.div>
`;
